### PR TITLE
modify achieving level of task logic to match technical report

### PR DESCRIPTION
**Pull request by maxeeem**
@bowen-xu I started going through the technical report and highlighting areas where PyNARS diverges. This is first of such PRs trying to align the implementation. Let me know if this works for you and we can discuss tomorrow during our sync up.

Screenshot 2023-10-25 at 10 51 32 AM

### CODE REVIEW

File: pynars/Narsese/_py/Task.py [hunk 1]
```diff
@@ -17,20 +17,13 @@ def __init__(self, sentence: Sentence, budget: Budget=None, input_id: int=None)
...
-        elif self.is_quest:
-            quest: Quest = self.sentence
-            return truth_belief.e if quest.is_query else truth_belief.c
+        if self.is_judgement or self.is_goal:
+            if truth_belief is not None:
+                return 1 - abs(self.truth.e - truth_belief.e)
+            else: # no previous belief
+                return abs(self.truth.e - 0.5)
+        elif self.is_question or self.is_quest:
+            return truth_belief.e if self.is_query else truth_belief.c
```
Comment (maxeeem) [inline pynars/Narsese/_py/Task.py:26]: @bowen-xu What should be returned if `truth_belief=None`? Should we return `0` in this case?

Comment (bowen-xu) [reply pynars/Narsese/_py/Task.py]: @maxeeem Good question... This is not explicitly claimed in the technical reprot. I'm not sure neither.

File: pynars/Narsese/_py/Task.py [hunk 2]
```diff
@@ -16,21 +16,22 @@ def __init__(self, sentence: Sentence, budget: Budget=None, input_id: int=None)
...
+        if self.is_judgement or self.is_goal:
+            if truth_belief is not None:
+                return 1 - abs(self.truth.e - truth_belief.e)
+            else: # no previous belief
+                return abs(self.truth.e - 0.5)
+        elif self.is_question or self.is_quest:
+            if truth_belief is not None:
+                return truth_belief.e if self.is_query else truth_belief.c
+            else: # no previous belief
+                return 0
```
Comment (bowen-xu) [inline pynars/Narsese/_py/Task.py:34]: A guess:
```
return 0.5 if self.is_query else 0
```

File: pynars/Narsese/_py/Task.py (hunk 1 above)
Comment (bowen-xu) [reply pynars/Narsese/_py/Task.py]: @maxeeem I discussed it with Pei. your current code is correct. will explain it later.

Comment (bowen-xu) [reply pynars/Narsese/_py/Task.py]: Achieving level of a question/quest means to what extent the it is solved. When there's no belief/goal, the extent is 0.

Comment (maxeeem) [reply pynars/Narsese/_py/Task.py]: @bowen-xu so is this PR ready to merge or some changes are still needed?

Comment (bowen-xu) [reply pynars/Narsese/_py/Task.py]: merge it now.

**Pull request merged**